docs/proposals/27-falco-extension.md

```diff
@@ -0,0 +1,318 @@
+
+```
+kind: ConfigMap
+  name: my-custom-rules
+  namespace: garden-myproject
+data:
+  myrules.yaml: |-
+    # custom rules go here
+    ...
+```
+
+Events that are generated by Falco are forwarded to Falcosidekick that is 
+deployed alongside Falco. The Falcosidekick webhook output is configured
+to send events to a central location. The Falco extension issues JWT
+tokens containing the cluster identity.
+
+As an alternative users can opt to send events to a custom location by
+defining a `customWebhook` key and configuration. In this case no events 
+will be forwarded to the default event ingestors, events will not be 
+visible in the Gardener dashboard, and there will be no alerting.
+
+#### Falco Versioning
+
+New Falco releases come with new features and might also discontinue existing
+features. Rule sets are usually bound to a particular Falco version. This
+means that updating Falco and/or rules may either break Falco, generate 
+false positive events or might not generate expected events anymore. 
+
+For these reasons we have therefore decided to offer multiple Falco versions
+allow limited control over which version to choose and provide time for 
+necessary updates. We want to roughly follow the shoot versioning policy 
+set out in [7], however with simplifications as the policy has proven to 
+be very complex especially in the operating system area. 
+
+We plan to implement the versioning policy with the following simplifications:
+
+- A linear update path without differentiation of major, minor, and patch 
+  versions.
+- Automated updates always update to the latest supported version.
+- Forced updates always update to the latest supported version.
+- There may be multiple supported versions.
+
+#### Falco Rules
+
+The Gardener extension will deploy matching rules for the Falco release from
+[8]. `falco_rules.yaml` will be deployed by default and Gardener will bring 
+extensions to the rule set to remediate false positives if necessary. 
+`falco-incubating_rules.yaml` and `falco-sandbox_rules.yaml` will only be 
+deployed if selected and there will be no exceptions by Gardener for 
+false positve events.
+
+#### Falcoctl
+
+Users with many clusters requested support for falcoctl [9]. There is 
+extra effort for configuration as an external repository for rules files
+and plug-ins is required for the benefit of controlling rules and 
+plugins centrally. We plan to pass the falcoctl configuration and disable 
+all Gardener managed rules for this use case.
+
+#### Falco Event Ingestor
+
+The Falco event ingestor receives Falco events from Falcosidekick, verifies
+the JWT token as well as the event body and stores it into an event store
+(currently a Postgres database). Due to receiving events from potentially 
+hundreds of clusters the Falco event ingestor must be highly scalable as
+well as to provide additional hardening including the following checks:
+
+- event rate limiting based on cluster identities
```

Comment:
The rate limit will be quite high (several thousands of events) and should never be hit during "normal" operation. The "attacker gains access to the jwt token" is the main threat that the rate limiting should protect against; otherwise the attacker could fill up the event store with garbage events. Should we create alerts when this is happening? Probably yes, but I need to think how this could be done.